### PR TITLE
Feat: Trending Events

### PR DESCRIPTION
**Pull request by KarimJC**
# Description
[Link to Ticket](https://github.com/GenerateNU/skillspark/issues/163)
Please include a summary of the changes and the related issue. Please also
include relevant motivation, context, and images!

This ticket added a trending events route to the backend. Trending events are decided by which event occurrences within 5k m have had the most registrations in the last 14 days. 

# How Has This Been Tested?
Please describe the tests that you manually ran to verify your changes (beyond any unit/integration tests written and ran).

## Screenshots
Please provide screenshots of manual testing (Scalar, frontend pages, etc.)

# Checklist

## General
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas

## Backend Changes (if applicable)
- [ ] I have written unit tests for my code
- [ ] I have written integration tests where appropriate
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have run make api-validate and make api-gen to keep the openAPI spec up to date
- [ ] I have tested error handling and edge cases
- [ ] Database migrations are included (if applicable)

## Frontend Changes (if applicable)
- [ ] I have tested in multiple browsers (Chrome, Firefox, Safari)
- [ ] Mobile responsive design verified (if applicable)
- [ ] I have run make generate-api to keep the API hooks up to date
- [ ] **For mobile features:** Tested on both iOS and Android devices
- [ ] **For mobile features:** Screenshots from both iOS and Android included below
- [ ] No console errors or warnings
- [ ] Loading states and error states handled

### iOS
[Add iOS screenshots here]
image

### Android
[Add Android screenshots here]

### CODE REVIEW

**Comment by KarimJC** [comment]
Only things that are not working are guardian login tests, which are also failing on main

File: backend/internal/supabase/migrations/20260407011738_haversine_function_addition.sql
Comment (adescoteaux1) [inline backend/internal/supabase/migrations/20260407011738_haversine_function_addition.sql:1]: I dton think you are actually using this anywhere, can we either update the sql function to use it or remove the migration

File: frontend/apps/mobile/app/(app)/(tabs)/index.tsx [hunk 1]
```diff
@@ -106,6 +109,34 @@ export default function HomeScreen() {
...
+      max_returns: 5,
+    },
+    {
+      query: {
+        enabled: !!geoLocationLat && !!geoLocationLong,
+      },
+    },
+  );
+
+  console.log({ trendingResp, isError, error, trendingLoading });
```
Comment (adescoteaux1) [inline frontend/apps/mobile/app/(app)/(tabs)/index.tsx]: remove console.log

File: frontend/apps/mobile/package.json [hunk 2]
```diff
@@ -50,7 +50,8 @@
     "react-native-svg": "15.12.1",
     "react-native-web": "~0.21.0",
     "react-native-worklets": "0.5.1",
-    "use-debounce": "^10.1.0"
+    "use-debounce": "^10.1.0",
+    "react": "19.1.0"
```
Comment (adescoteaux1) [inline frontend/apps/mobile/package.json]: @josh-torre are we trying to keep this out

File: backend/internal/service/handler/auth/handler_test.go [hunk 3]
```diff
@@ -103,7 +103,6 @@ func TestHandler_GuardianLogin(t *testing.T) {
 	for _, tt := range tests {
 		tt := tt // capture range variable for parallel
 		t.Run(tt.name, func(t *testing.T) {
-			t.Parallel()
```
Comment (josh-torre) [inline backend/internal/service/handler/auth/handler_test.go:106]: Why did we remove these?

File: frontend/apps/mobile/package.json (hunk 2 above)
Comment (josh-torre) [reply frontend/apps/mobile/package.json]: Yes

Comment (josh-torre) [reply frontend/apps/mobile/package.json]: Please remove the dependency of react. Its already installed at a higher level.

File: backend/internal/service/handler/auth/handler_test.go (hunk 3 above)
Comment (KarimJC) [reply backend/internal/service/handler/auth/handler_test.go:106]: It fixed the test I believe. @adescoteaux1 did this

Comment (adescoteaux1) [reply backend/internal/service/handler/auth/handler_test.go:106]: @josh-torre parallelizing auth tests breaks them bc its using the same client. it would require semi major refactoring of the code to fix it otherwise

**Comment by josh-torre** [comment]
~~@KarimJC Please run bun install at the dependency so it gets removed from the bun.lock~~

**Review by josh-torre — APPROVED** [review]
